editor/src/ast/ast.rs

```diff
@@ -323,18 +338,24 @@ impl<'a, 'l> FlexibleAst<'a, 'l> {
     }
 }
 
-pub struct ReadText<'f, 'l>(pub(super) ReadLeaf<'f, Node<'l>, String>);
+pub struct ReadText<'f, 'l>(pub(super) ReadLeaf<'f, Node<'l>, Text>);
+
+impl<'f, 'l> ReadText<'f, 'l> {
+    pub fn as_text_ref(&self) -> &Text {
+        self.0.deref()
+    }
+}
```

Comment:
TODO: Make this `AsRef<Text>`, and make `AsRef<str>` instead be `IntoIterator<&char>` for both this and `PrettyDocument`.

Reply:
see https://github.com/justinpombrio/synless/tree/char-iterator-crap instead